Week-1/Readme.md

```diff
@@ -0,0 +1,58 @@
+
+**The rendering engine:** responsible for displaying requested content. For example if the requested content is HTML, the rendering engine parses HTML and CSS, and displays the parsed content on the screen.
+
+**Networking:** for network calls such as HTTP requests, using different implementations for different platform behind a platform-independent interface.
+
+**UI backend:** used for drawing basic widgets like combo boxes and windows. This backend exposes a generic interface that is not platform specific. Underneath it uses operating system user interface methods.
+
+**JavaScript interpreter:** Used to parse and execute JavaScript code.
+
+**Data storage :** This is a persistence layer. The browser may need to save all sorts of data locally, such as cookies. Browsers also support storage mechanisms such as localStorage, IndexedDB, WebSQL and FileSystem.
+
+![High Level Components of Browser](./images/HIghLevelComponent.png?raw=true "Title")
+
+## Rendering engine and its use
+Once a user requests a particular document, the rendering engine starts fetching the content of the requested document. This is done via the networking layer. The rendering engine starts receiving the content of that specific document in chunks of 8 KBs from the networking layer. After this, the basic flow of the rendering engine begins.
+
+The four basic steps include:
+
+1 - The requested HTML page is parsed in chunks, including the external CSS files and in style elements, by the rendering engine. The HTML elements are then converted into DOM nodes to form a “content tree” or “DOM tree.”
+
+2 - Simultaneously, the browser also creates a render tree. This tree includes both the styling information as well as the visual instructions that define the order in which the elements will be displayed. The render tree ensures that the content is displayed in the desired order.
+
+3 - Further, the render tree goes through the layout process. When a render tree is created, the position or size values are not assigned. The entire process of calculating values for evaluating the desired position is called a layout process. In this process, every node is assigned the exact coordinates. This ensures that every node appears at an accurate position on the screen.
+
+4 - The final step is to paint the screen, wherein the render tree is traversed, and the renderer’s paint() method is invoked, which paints each node on the screen using the UI backend layer.
+
+![Rendering Engine](./images/rendering%20engine.jpg?raw=true "Title")
+
+## Parsers (HTML, CSS, etc)
+Parsing means analyzing and converting a program into an internal format that a runtime environment can actually run, for example the JavaScript engine inside browsers. The browser parses HTML into a DOM tree. HTML parsing involves tokenization and tree construction.
+
+## Script Processors
+The script processor executes Javascript code to process an event.
```

Comment:
@porushmarwaha lets add more info on script execution and why we use async and defer. And why CSSOM and script tags are render blocking